packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTreeComponent.tsx

```diff
@@ -127,8 +127,11 @@ function ModelsTreeComponentImpl(props: ModelTreeComponentProps & { iModel: IMod
   const queryModels = useCallback(async (): Promise<ModelInfo[]> => {
     const queryParams: ModelQueryParams = {
       from: "BisCore.GeometricModel3d",
+      where: `GeometricModel3d.ModeledElement.Id IN (select ECInstanceId FROM BisCore.GeometricElement3d)
+      OR GeometricModel3d.ModeledElement.Id IN (select ECInstanceId FROM BisORe.InformationPartitionElement)`,
```

Comment:
The query plan of your query is:
```txt
id      parent  notused detail
-------------------------------------------------------------
7       0       0       MULTI-INDEX OR
8       7       0       INDEX 1
13      8       0       LIST SUBQUERY 3
15      13      0       SCAN main.bis_GeometricElement3d USING COVERING INDEX ix_bis_GeometricElement3d_fk_bis_GeometricElement3dIsInCategory_source
30      8       0       SEARCH main.bis_Model USING INDEX uix_bis_Model_fk_bis_ModelModelsElement_source (ModeledElementId=?)
36      7       0       INDEX 2
41      36      0       LIST SUBQUERY 5
43      41      0       SCAN main.bis_InformationPartitionElement USING COVERING INDEX ix_bis_InformationPartitionElement_ecclassid
58      36      0       SEARCH main.bis_Model USING INDEX uix_bis_Model_fk_bis_ModelModelsElement_source (ModeledElementId=?)
69      0       0       LIST SUBQUERY 3
71      69      0       SCAN main.bis_GeometricElement3d USING COVERING INDEX ix_bis_GeometricElement3d_fk_bis_GeometricElement3dIsInCategory_source
89      0       0       LIST SUBQUERY 5
91      89      0       SCAN main.bis_InformationPartitionElement USING COVERING INDEX ix_bis_InformationPartitionElement_ecclassid
104     0       0       SEARCH CHC_bis_Model USING INDEX ix_ec_cache_ClassHierarchy_ClassId (ClassId=?)
```

I ran it on 4 of my largest iModels, the query took (in seconds): 5.5, 10.5, 0.1, 5.6. The below suggested query on the same iModels takes: 0.05  / 0.1  / 0.03 / 0.004. Its query plan:

```txt
id      parent  notused detail
-------------------------------------------------------------
6       0       0       SEARCH CHC_bis_Model USING INDEX ix_ec_cache_ClassHierarchy_BaseClassId (BaseClassId=?)
11      0       0       SEARCH main.bis_Model USING INDEX ix_bis_Model_ecclassid (ECClassId=?)
18      0       0       CORRELATED SCALAR SUBQUERY 4
22      18      0       SEARCH main.bis_Element USING INTEGER PRIMARY KEY (rowid=?)
29      18      0       LIST SUBQUERY 3
32      29      0       SEARCH ec_cache_ClassHierarchy USING INDEX ix_ec_cache_ClassHierarchy_BaseClassId (BaseClassId=?)
```

```suggestion
      where: `
        EXISTS (
          SELECT 1
          FROM BisCore.Element e
          WHERE e.ECClassId IS (BisCore.GeometricElement3d, BisCore.InformationPartitionElement)
            AND e.ECInstanceId = GeometricModel3d.ModeledElement.Id
        )
      `,
```